phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -3416,6 +3437,167 @@ private MetaDataMutationResult mutateColumn(
+        List<CompletableFuture<Void>> futures = new ArrayList<>();
+        Map<Future, ServerName> map = new HashMap<>();
+        for (ServerName serverName : serverNames) {
+            CompletableFuture<Void> future = CompletableFuture.runAsync(() -> {
+                try {
+                    PhoenixStopWatch innerWatch = new PhoenixStopWatch().start();
+                    // TODO Using the same as ServerCacheClient but need to think if we need some
+                    // special controller for invalidating cache since this is in the path of
+                    // DDL operations. We also need to think of we need separate RPC handler
+                    // threads for this?
+                    ServerRpcController controller = new ServerRpcController();
+                    RegionServerEndpointProtos.RegionServerEndpointService.BlockingInterface
+                            service = RegionServerEndpointProtos.RegionServerEndpointService
+                            .newBlockingStub(admin.coprocessorService(serverName));
+                    LOGGER.info("Sending invalidate metadata cache for tenantID: {}, tableName: {}"
+                            + " to region server: {}", tenantIDStr, fullTableName, serverName);
+                    // The timeout for this particular request is managed by config parameter:
+                    // hbase.rpc.timeout. Even if the future times out, this runnable can be in
+                    // RUNNING state and will not be interrupted.
+                    service.invalidateServerMetadataCache(controller, request);
+                    LOGGER.info("Invalidating metadata cache for tenantID: {}, tableName: {}"
+                            + " on region server: {} completed successfully and it took {} ms",
+                            tenantIDStr, fullTableName, serverName,
+                            innerWatch.stop().elapsedMillis());
+                    // TODO Create a histogram metric for time taken for invalidating the cache.
+                } catch (ServiceException se) {
+                    LOGGER.error("Invalidating metadata cache for tenantID: {}, tableName: {}"
+                                    + " failed for regionserver {}", tenantIDStr, fullTableName,
+                            serverName, se);
+                    IOException ioe = ServerUtil.parseServiceException(se);
+                    throw new CompletionException(ioe);
+                }
+            });
+            futures.add(future);
+            map.put(future, serverName);
+        }
+
+        // Here we create one master like future which tracks individual future
+        // for each region server.
+        CompletableFuture<Void> allFutures = CompletableFuture.allOf(
+                futures.toArray(new CompletableFuture[0]));
+        try {
+            allFutures.get(metadataCacheInvalidationTimeoutMs, TimeUnit.MILLISECONDS);
+        } catch (Throwable t) {
```

Comment:
This catches every kind of exception, is it desirable to retry in error cases? Why not be specific to those that are just declared?
```suggestion
        } catch (CancellationException | ExecutionException | InterruptedException | TimeoutException t) {
```

Reply:
In case of non declared exception, what do you think we should do? Should we just fail the DDL operation and not retry? 
I was thinking to catch all the exception, retry once and if it still fails then fail the operation.